sound/soc/sof/ipc4-topology.c

```diff
@@ -1535,7 +1547,11 @@ static int sof_ipc4_widget_setup(struct snd_sof_dev *sdev, struct snd_sof_widget
 
 		msg->extension &= ~SOF_IPC4_MOD_EXT_PARAM_SIZE_MASK;
 		msg->extension |= ipc_size >> 2;
+
+		msg->extension &= ~SOF_IPC4_MOD_EXT_PPL_ID_MASK;
+		msg->extension |= SOF_IPC4_MOD_EXT_PPL_ID(pipe_widget->instance_id);
```

Comment:
I am confused about this. Isn't the need to set the PPL_ID information orthogonal with how the pipeline numbering is done? Shouldn't this be a separate patch to improve/harden the existing IPC4?

Reply:
@bardliao you need to remove the pipeline ID seting on line 332 and like Pierre suggested do this in a separate patch

Reply:
> @bardliao you need to remove the pipeline ID seting on line 332 and like Pierre suggested do this in a separate patch

done

Reply:
@plbossart @ranj063 I don't think it belongs to a separate patch. The existing code use a fixed pipeline id which is assigned when loading topology and will not be changed. The patch needs this because we assign instance_id dynamically.

Reply:
I don't get this either. Even if we used a fixed setting, why didn't we need the message extension before.

My point is that regardless of what value is used for the pipeline, the sequence is new, and I fail to see why it was not needed before.

Reply:
@plbossart we did use the message extension. It was done in sof_ipc4_widget_setup_msg() which is called before the pipeline widget is set up in the firmware with the pipeline ID that was statically allocated in topology. But now since it is allocated dynamically, this has been moved to  sof_ipc4_widget_setup() which comes after the pipeline widget is set up and the dynamic pipeline ID is available.

Reply:
@ranj063 I only see an addition in this patch, not a code move. What am I missing?

Reply:
@plbossart please see the removal of line 331/332 above